webapp/components/FilterMenu/OrderByFilter.vue

```diff
@@ -36,15 +36,69 @@ export default {
     FilterMenuSection,
     LabeledButton,
   },
+  mixins: [FilterMenuMixin],
   computed: {
-    ...mapGetters({
-      orderBy: 'posts/orderBy',
-    }),
+    orderedAsc() {
+      return this.orderedByCreationDate ? 'createdAt_asc' : 'eventStart_desc'
+    },
+    orderedDesc() {
+      return this.orderedByCreationDate ? 'createdAt_desc' : 'eventStart_asc'
+    },
+    sectionTitle() {
+      return this.orderedByCreationDate
+        ? this.$t('filter-menu.creationDate')
+        : this.$t('filter-menu.startDate')
+    },
   },
   methods: {
-    ...mapMutations({
-      toggleOrder: 'posts/TOGGLE_ORDER',
-    }),
+    buttonLable(buttonType) {
+      let title
+      switch (buttonType) {
+        case 'asc':
+          if (this.orderedByCreationDate) {
+            title = this.$t('filter-menu.order.oldest.label')
+          } else {
+            title = this.$t('filter-menu.order.last.label')
+          }
+          break
+        case 'desc':
+          if (this.orderedByCreationDate) {
+            title = this.$t('filter-menu.order.newest.label')
+          } else {
+            title = this.$t('filter-menu.order.next.label')
+          }
+          break
+
+        default:
+          title = ''
+          break
+      }
+      return title
```

Comment:
Not mega fan of this kind of constructs.
```suggestion
      switch (buttonType) {
        case 'asc':
          return this.orderedByCreationDate ? this.$t('filter-menu.order.oldest.label') : this.$t('filter-menu.order.last.label')
        case 'desc':
          return this.orderedByCreationDate ? this.$t('filter-menu.order.newest.label') : this.$t('filter-menu.order.next.label')
        default:
          return ''
      }
```